wrapper/src/main/java/software/amazon/jdbc/plugin/strategy/fastestresponse/FastestResponseStrategyPlugin.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.strategy.fastestresponse;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import software.amazon.jdbc.AwsWrapperProperty;
+import software.amazon.jdbc.HostRole;
+import software.amazon.jdbc.HostSpec;
+import software.amazon.jdbc.JdbcCallable;
+import software.amazon.jdbc.NodeChangeOptions;
+import software.amazon.jdbc.PluginService;
+import software.amazon.jdbc.PropertyDefinition;
+import software.amazon.jdbc.RandomHostSelector;
+import software.amazon.jdbc.plugin.AbstractConnectionPlugin;
+import software.amazon.jdbc.util.CacheMap;
+
+public class FastestResponseStrategyPlugin extends AbstractConnectionPlugin {
+
+  private static final Logger LOGGER =
+      Logger.getLogger(FastestResponseStrategyPlugin.class.getName());
+
+  public static final String FASTEST_RESPONSE_STRATEGY_NAME = "fastestResponse";
+
+  private static final Set<String> subscribedMethods =
+      Collections.unmodifiableSet(new HashSet<String>() {
+        {
+          add("notifyNodeListChanged");
+          add("acceptsStrategy");
+          add("getHostSpecByStrategy");
+        }
+      });
+
+  public static final AwsWrapperProperty RESPONSE_MEASURE_INTERVAL =
+      new AwsWrapperProperty(
+          "responseMeasureInterval",
+          "30000",
+          "Interval in millis between measuring response time to a database node.");
+
+  protected static final CacheMap<String, HostSpec> latestResponseHost = new CacheMap<>();
```

Comment:
latestResponseMap ? or latestResponseCache?

the current name makes it looks like it's a host